lib/ht_sip_validator/validation/checksums.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+require "ht_sip_validator/validation/base"
+
+module HathiTrust
+  module Validation
+    module Checksums
+      # validates that package contains checksums
+      class Exists < HathiTrust::Validation::Base
+        def validate
+          unless @sip.files.include?(HathiTrust::SIP::CHECKSUM_FILE)
+            @messages << Message.new(
+              validator: self.class,
+              validation: :exists,
+              level: Message::ERROR,
+              human_message: "SIP is missing #{HathiTrust::SIP::CHECKSUM_FILE}",
+              extras: { filename: HathiTrust::SIP::CHECKSUM_FILE }
+            )
+          end
+
+          super
+        end
+      end
+
+      # validates that checksums exist for all files in the sip
+      class FileListComplete < HathiTrust::Validation::Base
+        def validate
+          @sip.files.each do |filename|
```

Comment:
Emitting a warning for things like `.DS_Store` and `Thumbs.db` is probably preferable